### ISSUE TITLE
Secure boot

### ISSUE THREAD

**Issue by komachi** [komachi/ansible-decent-desktop#2]
With deploying your own keys and signing everything with own keys, including dkms modules.

**Comment by savchenko** [comment]
https://github.com/osresearch/safeboot ?

**Comment by savchenko** [comment]
Oh, and this one: https://github.com/Foxboron/sbctl

**Comment by komachi** [comment]
Bootstraping should be covered with [sbkeysync](https://manpages.debian.org/bullseye/sbsigntool/sbkeysync.1.en.html). sbctl looks nice, but I am trying to avoid deps outside of debian as well as I can.

For DKMS signing there is `sign_tool=` option in `/etc/dkms/framework.conf` which should run something like `/lib/modules/"$1"/build/scripts/sign-file sha512 /root/mok.priv /root/mok.der "$2"`, so DKMS modules can be automatically signed on install.

For kernel upgrades we can install custom postinst hook, something like [this](https://gist.github.com/maxried/796d1f3101b3a03ca153fa09d3af8a11). I found also [sicherboot](https://github.com/julian-klode/sicherboot) for inspiration, it doubles sbkeysync in some features, but shows how to sign efi kernel image.

**Comment by savchenko** [comment]
> Bootstraping should be covered with sbkeysync

Interesting find! Had no idea it exists... Which is probably unsurprising given it's available only in `unstable`.

Are you planning to wrap SB-related steps into some sort of automated or guided workflow? Ansible playbook, shell script, etc.

**Comment by komachi** [comment]
Yes. this requires sbsigntool from unstable, hope it get's backported soon. It doesn't depends on anything besides libc, libssl, and libuuid, so this could be installed on bullseye without breaking much even in case if it doesn't get backported.

Yes, I plan to manage SB with ansible. There is some (not tested and unused for now) draft for moving to systemd-boot in `roles/boot_manager`,  so this will be updated to support secureboot.
